updater/spec/dependabot/job_spec.rb

```diff
@@ -63,6 +63,30 @@
   let(:commit_message_options) { nil }
   let(:vendor_dependencies) { false }
 
+  describe "::new_update_job" do
+    let(:job_json) { fixture("jobs/job_with_credentials.json") }
```

Comment:
[Fixture](https://github.com/dependabot/dependabot-core/blob/ce38f0d9b5e6004eb97041a25157ce6332bfc7c7/updater/spec/fixtures/jobs/job_with_credentials.json)

This file includes both `credentials` and `credentials-metadata` which isn't representative of the kind of files we actually inject, but it is a good way to make sure we use the metadata by precedent if both are present.